### ISSUE TITLE
Travis CI flakes

### ISSUE THREAD

**Issue by gvanrossum** [python/mypy#1491]
We've seen a few times recently that the Travis CI build fails with a strange EOFError while importing one of mypy's own modules (e.g. mypy/traverser.py). So far only (or mostly?) in 3.2 builds. Restarting the build will usually succeed. And mostly on the cmdline tests (but not always).

Theory: maybe now that we have separate cmdline and python-eval tasks these are running in parallel using the same temp directory?

**Comment by gvanrossum** [comment]
Example: https://travis-ci.org/python/mypy/jobs/128328958; excerpt:
```
$ python runtests.py -v
PARALLEL 2
SUMMARY  198 tasks selected
START    #0 run eval-test
START    #1 run cmdline-test
EXIT 1   #1 run cmdline-test
FAILURE  #1 run cmdline-test
Expected:
  pkg/a.py:1: error: Name 'undef' is not defined (diff)
  pkg/subpkg/a.py:1: error: Name 'undef' is not defined (diff)
Actual:
  Traceback (most recent call last):            (diff)
    File "/home/travis/build/python/mypy/scripts/mypy", line 4, in <module> (diff)
      from mypy.main import main                (diff)
    File "/home/travis/build/python/mypy/mypy/main.py", line 10, in <module> (diff)
      from mypy import build                    (diff)
    File "/home/travis/build/python/mypy/mypy/build.py", line 30, in <module> (diff)
      from mypy.checker import TypeChecker      (diff)
    File "/home/travis/build/python/mypy/mypy/checker.py", line 37, in <module> (diff)
      import mypy.checkexpr                     (diff)
    File "/home/travis/build/python/mypy/mypy/checkexpr.py", line 25, in <module> (diff)
      from mypy.replacetvars import replace_func_type_vars (diff)
  EOFError: EOF read where not expected         (diff)
Alignment of first line difference:
  E: pkg/a.py:1: error: Name 'undef' is not defined
  A: Traceback (most recent call last):
     ^
Traceback (most recent call last):
  File "mypy/test/data.py", line 141, in run
    self.perform(self)
  File "mypy/test/testcmdline.py", line 60, in test_python_evaluation
    testcase.file, testcase.line))
  File "mypy/test/helpers.py", line 86, in assert_string_arrays_equal
    raise AssertionFailure(msg)
AssertionFailure: Invalid output (mypy/test/data/cmdline.test, line 12)
test_testcmdline_PythonEvaluationSuite.testCmdlinePackage failed
1/6 test cases failed.
*** FAILURE ***
```

**Comment by gvanrossum** [comment]
A little Googling showed that this error ("EOFError: EOF read where not expected") means a bad .pyc file was read. My guess is that the following sequence of events is happening, where three (!) processes A, B and C both import the same module:

- A reads foo.py
- B reads foo.py
- A writes foo.pyc
- C starts reading foo.pyc
- B overwrites foo.pyc
- C finds an early EOF

I guess the three (or more) processes are some combination of the runtest subprocesses used for task parallelization (there are 2 of these) and the mypy subprocesses used to run the tests (one per task).

We're only seeing this with Python 3.2 since it was fixed in 3.3: http://bugs.python.org/issue13146

Since Python 3.2 is a lame duck anyway and this seems hard to fix otherwise I propose to just live with it. (If it gets too annoying I can change cmdline-test to eval-test and then IIUC they are all run in the same task, but that seems mildly wrong too -- that task is already one of the slowest.)

**Comment by gvanrossum** [comment]
(Labeling with New Parser since we can close this once the new parser is the default and 3.2 is a goner.)
